test/functional/tests/fault_injection/test_max_io_greater_in_core.py

```diff
@@ -0,0 +1,96 @@
+          various CAS configurations.
+        pass_criteria:
+          - No kernel bug.
+          - Running workload successfully.
+          - Fulfilling criteria: core.max_io > cache.max_io & core.max_io > 1MiB.
+    """
+    with TestRun.step("Prepare devices."):
+        core_scsi_debug_disk = TestRun.scsi_debug_devices[0]
+
+        cache_disk = TestRun.disks['cache']
+        cache_disk.create_partitions([Size(1, Unit.GibiByte)])
+        cache_dev = cache_disk.partitions[0]
+
+        TestRun.executor.run(f"mount /DVD")
+
+    with TestRun.step("Set 'max_sectors_kb' for core device"):
+        max_hw_io_size = core_scsi_debug_disk.get_max_hw_io_size()
+        new_max_io_size = min(max_hw_io_size, Size(4, Unit.MebiByte))
+        core_scsi_debug_disk.set_max_io_size(new_max_io_size)
+
+    with TestRun.step("Check if max_sectors_kb for cache device is less than core's"):
+        core_max_io = core_scsi_debug_disk.get_max_io_size()
+        cache_max_io = cache_disk.get_max_io_size()
+        TestRun.LOGGER.info(f"Cache max io size: {cache_max_io}, Core max io size: {core_max_io}")
+
+        if core_max_io <= cache_max_io or core_max_io < Size(1000, Unit.KiloByte):
+            TestRun.LOGGER.error("Could not set core or cache max_sectors_kb limit - "
+                                 "Assumption for core.max_io > cache.max_io & core.max_io > 1MiB. "
+                                 f"core.max_io value is {core_max_io}")
+
+    with TestRun.step("Unload CAS modules if needed"):
+        if os_utils.is_kernel_module_loaded(module.value):
+            cas_module.unload_all_cas_modules()
+
+    with TestRun.step(f"Loading CAS modules"):
+        os_utils.load_kernel_module(cas_module.CasModule.cache.value)
+
+    with TestRun.step("Starting cache"):
+        cache = casadm.start_cache(cache_dev, cache_mode, cache_line_size, force=True)
+
+    with TestRun.step("Adding core"):
+        core = cache.add_core(core_scsi_debug_disk)
+
+    with TestRun.step("Run IO"):
+        block_size_multipliers = [1, 2, 4, 8]
+        seek_bytes = Size(0, Unit.Byte)
+
+        for block_size_multiplier in block_size_multipliers:
+
+            block_size = core_max_io * block_size_multiplier
+
+            TestRun.LOGGER.info(f"Testing with block size: {block_size}")
+            dd = Dd().input("/dev/zero") \
+                .output(core) \
+                .count(20) \
+                .block_size(block_size) \
+                .oflag("direct") \
+                .seek(seek_bytes / block_size)
+            dd.run()
+
+            seek_bytes += block_size * 20   # each iteration will cover different disk part
+
+            casadm.stop_cache(cache.cache_id)
+            casadm.load_cache(cache_dev)
+
+        with TestRun.step(f"Stop CAS"):
+            casadm.stop_cache(cache.cache_id)
+
+        with TestRun.step(f"Unloading CAS modules"):
+            cas_module.unload_all_cas_modules()
```

Comment:
Why are you unloading CAS modules at the end of the test?

Reply:
unnecessary